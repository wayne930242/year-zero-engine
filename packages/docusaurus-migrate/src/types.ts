/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export type RawData = {
  header?: string;
  content?: string;
};

export type Data = {
  metadata: {[key: string]: string};
  rawContent: string;
};

export type ClassicPresetEntries = {
  docs: {[key: string]: any};
  blog: {[key: string]: any};
  theme: {[key: string]: any};
};

export type SidebarEntries = {
  [key: string]:
    | Record<string, unknown>
    | Array<Record<string, unknown> | string>;
};

export interface VersionTwoConfig {
  baseUrl: string;
  favicon: string;
  tagline?: string;
  title: string;
  url: string;
  organizationName?: string;
  projectName?: string;
  noIndex?: boolean;
  githubHost?: string;
  onBrokenLinks: string;
  onBrokenMarkdownLinks: string;
  plugins: Array<[string, {[key: string]: any}]>;
  themes?: [];
  presets: [[string, ClassicPresetEntries]];
  themeConfig: {
    gtag?: {
      trackingID?: string;
    };
    navbar: {
      title?: string;
      logo?: {
        src?: string;
      };
      items: Array<Record<string, unknown> | null>;
    };
    image?: string;
    footer: {
      links: Array<{
        title: string;
        items: Array<{
          label: string;
          to: string;
        }>;
      }>;
      copyright?: string;
      logo: {
        src?: string;
      };
    };
    algolia?: Record<string, unknown>;
  };
  customFields: {
    [key: string]: unknown;
  };
  scripts?: (
    | string
    | {
        src: string;
        [key: string]: unknown;
      }
  )[];
  stylesheets?: (
    | string
    | {
        href: string;
        [key: string]: unknown;
      }
  )[];
}

export type VersionOneConfig = {
  title?: string;
  tagline?: string;
  url?: string;
  baseUrl?: string;
  defaultVersionShown?: string;
  organizationName?: string;
  projectName?: string;
  noIndex?: boolean;
  headerLinks?: Array<any>;
  headerIcon?: string;
  favicon?: string;
  colors?: any;
  copyright?: string;
  editUrl?: string;
  customDocsPath?: string;
  users?: Array<Record<string, unknown>>;
  disableHeaderTitle?: string;
  disableTitleTagline?: string;
  separateCss?: Array<Record<string, unknown>>;
  footerIcon?: string;
  translationRecruitingLink?: string;
  algolia?: Record<string, unknown>;
  gaTrackingId?: string;
  highlight?: Record<string, unknown>;
  markdownPlugins?: Array<() => void>;
  scripts?: Array<{src: string; [key: string]: any} | string>;
  stylesheets?: Array<{href: string; [key: string]: any} | string>;
  facebookAppId?: string;
  facebookComments?: true;
  facebookPixelId?: string;
  twitter?: string;
  twitterUsername?: string;
  twitterImage?: string;
  ogImage?: string;
  cleanUrl?: boolean;
  scrollToTop?: boolean;
  scrollToTopOptions?: Record<string, unknown>;
};
