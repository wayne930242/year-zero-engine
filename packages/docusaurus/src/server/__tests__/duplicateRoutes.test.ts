/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import {
  getAllDuplicateRoutes,
  getDuplicateRoutesMessage,
  handleDuplicateRoutes,
} from '../duplicateRoutes';
import {RouteConfig} from '@docusaurus/types';

const routes: RouteConfig[] = [
  {
    path: '/',
    component: '',
    routes: [
      {path: '/search', component: ''},
      {path: '/sameDoc', component: ''},
    ],
  },
  {
    path: '/',
    component: '',
    routes: [
      {path: '/search', component: ''},
      {path: '/sameDoc', component: ''},
      {path: '/uniqueDoc', component: ''},
    ],
  },
];

describe('duplicateRoutes', () => {
  test('getDuplicateRoutesMessage', () => {
    const message = getDuplicateRoutesMessage([
      '/',
      '/',
      '/blog',
      '/doc/search',
    ]);
    expect(message).toMatchSnapshot();
  });

  test('getAllDuplicateRoutes', () => {
    expect(getAllDuplicateRoutes(routes)).toEqual(['/search', '/sameDoc']);
  });

  test('handleDuplicateRoutes', () => {
    expect(() => {
      handleDuplicateRoutes(routes, 'throw');
    }).toThrowErrorMatchingSnapshot();
  });
});
