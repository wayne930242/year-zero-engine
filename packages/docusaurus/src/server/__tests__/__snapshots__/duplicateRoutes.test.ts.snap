// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`duplicateRoutes getDuplicateRoutesMessage 1`] = `
"Attempting to create page at /, but a page already exists at this route
Attempting to create page at /, but a page already exists at this route
Attempting to create page at /blog, but a page already exists at this route
Attempting to create page at /doc/search, but a page already exists at this route"
`;

exports[`duplicateRoutes handleDuplicateRoutes 1`] = `
"Duplicate routes found!
Attempting to create page at /search, but a page already exists at this route
Attempting to create page at /sameDoc, but a page already exists at this route
This could lead to non-deterministic routing behavior"
`;
